ca/ca_test.go

```diff
@@ -912,13 +915,70 @@ func (qsa *queueSA) AddCertificate(_ context.Context, _ []byte, _ int64, _ []byt
 }
 
 func (qsa *queueSA) AddPrecertificate(ctx context.Context, req *sapb.AddCertificateRequest) (*corepb.Empty, error) {
-	return &corepb.Empty{}, nil
+	if qsa.fail {
+		return nil, errors.New("bad")
+	} else if qsa.duplicate {
+		return nil, berrors.DuplicateError("is a dupe")
+	}
+	issued := time.Unix(0, *req.Issued)
+	qsa.issuedPrecert = &issued
+	return nil, nil
 }
 
 func (qsa *queueSA) AddSerial(ctx context.Context, req *sapb.AddSerialRequest) (*corepb.Empty, error) {
 	return &corepb.Empty{}, nil
 }
 
+func TestPrecertOrphanQueue(t *testing.T) {
```

Comment:
I think this test would benefit from some higher level comments about what is being tested and why.